tests/functional/aws-node-sdk/test/object/rangeTest.js

```diff
@@ -0,0 +1,254 @@
+                    }
+                    return s3.uploadPartAsync({
+                        Bucket: bucket,
+                        Key: key,
+                        PartNumber: n + 1, // An MPU part cannot be 0
+                        UploadId: uploadId,
+                        Body: fs.createReadStream(`${name}.mpuPart${n + 1}`),
+                    })
+                    .then(() => next())
+                    .catch(next);
+                });
+        }, err => {
+            if (err) {
+                return s3.abortMultipartUploadAsync({
+                    Bucket: bucket,
+                    Key: key,
+                    UploadId: uploadId,
+                })
+                .then(() => cb(err));
+            }
+            return completeMPU(range, fileSize, uploadId, cb);
+        }));
+
+describe('aws-node-sdk range test for large end position', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        s3 = bucketUtil.s3;
+
+        beforeEach(done => {
+            s3.createBucketAsync({ Bucket: bucket })
+            .then(() => createHashedFile(2890, done))
+            .catch(done);
+        });
+
+        afterEach(done =>
+            bucketUtil.empty(bucket)
+            .then(() => bucketUtil.deleteOne(bucket))
+            .then(() => exec('rm getRangeExec hashedFile.*', done))
+            .catch(done)
+        );
+
+        it('should get the final 90 bytes of a 2890 byte object for a byte ' +
+            'range of 2800-', done => putRangeTest('2800-', 2890, done));
+
+        it('should get the final 90 bytes of a 2890 byte object for a byte ' +
+            'range of 2800-Number.MAX_SAFE_INTEGER', done =>
+            putRangeTest(`2800-${Number.MAX_SAFE_INTEGER}`, 2890, done));
+    });
+});
+
+describe('aws-node-sdk range test of regular object put (non-MPU)', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        s3 = bucketUtil.s3;
+
+        beforeEach(done => {
+            s3.createBucketAsync({ Bucket: bucket })
+            .then(() => createHashedFile(200, done))
+            .catch(done);
+        });
+
+        afterEach(done =>
+            bucketUtil.empty(bucket)
+            .then(() => bucketUtil.deleteOne(bucket))
+            .then(() => exec('rm getRangeExec hashedFile.*', done))
+            .catch(done)
+        );
+
+        it('should get a range for an object put without MPU', done =>
+            putRangeTest('10-99', 200, done));
+
+        it('should get a range for an object using only an end offset in the ' +
+            'request', done => putRangeTest('-10', 200, done));
+
+        it('should get a range for an object using only a begin offset in the' +
+            'request', done => putRangeTest('190-', 200, done));
+
+        it('should get full object if range header is invalid', done =>
+            putRangeTest('-', 200, done));
```

Comment:
I've increased the file size to 2000 bytes to add some additional tests at particular intervals, as based on the sample.